Utilla/GamemodeSelector.cs

```diff
@@ -67,7 +68,7 @@ GameObject CreatePageButton(string text, Action onPressed)
 
 				if (!button.GetComponentInParent<Canvas>())
 				{
-					Canvas canvas = button.transform.parent.gameObject.AddComponent<Canvas>();
+					Canvas canvas = button.transform.parent.gameObject.GetOrAddComponent<Canvas>();
```

Comment:
Doesn't `!button.GetComponentInParent<Canvas>()` guarantees that `GetOrAddComponent` will always be the same as `AddComponent`?

Reply:
i thought of that too, but without the code i had there would be errors thrown for i'm pretty sure mountain clouds and beach showing that a canvas already exists

Reply:
Huh weird, I got the same error though